### PR TITLE
feat: Added support for timestamp in log entry

### PR DESCRIPTION
**Pull request by MMaiero**
Signed-off-by: Maiero <matteo.maiero@eurotech.com>

> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

### CODE REVIEW

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]
In https://github.com/eclipse/kura/blob/14773ce93f3a724fc4a203ecbb1ede8883dc740f/kura/org.eclipse.kura.log.filesystem.provider/src/main/java/org/eclipse/kura/log/filesystem/provider/KuraLogLineParser.java#L116 we could set the timestamp in the constructor of `LogEntry`.

In the web UI, we could also use `LogEntry.getTimestamp` in the creation of the `GwtLogEntry` https://github.com/eclipse/kura/blob/14773ce93f3a724fc4a203ecbb1ede8883dc740f/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtLogServiceImpl.java#L72

Such class will need some modifications like using the passed timestamp for creating the "pretty" message https://github.com/eclipse/kura/blob/14773ce93f3a724fc4a203ecbb1ede8883dc740f/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtLogEntry.java#L102

File: kura/org.eclipse.kura.log.filesystem.provider/src/main/java/org/eclipse/kura/log/filesystem/provider/KuraLogLineParser.java [hunk 1]
```diff
@@ -106,13 +110,20 @@ private void parseKuraAuditLog() {
...
         entryProperties.put("_PID", instance.pid);
         entryProperties.put("MESSAGE", instance.message);
         entryProperties.put("PRIORITY", instance.priority);
         entryProperties.put("SYSLOG_IDENTIFIER", instance.syslogIdentifier);
         entryProperties.put("_TRANSPORT", instance.transport);
         entryProperties.put("STACKTRACE", instance.stacktrace);
+        
+        if (!DEFAULT_TIMESTAMP.equals(instance.timestamp)) {
+            try {
+                return new LogEntry(entryProperties, Long.parseLong(instance.timestamp.substring(0, 13)));
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.log.filesystem.provider/src/main/java/org/eclipse/kura/log/filesystem/provider/KuraLogLineParser.java]: Kura log timestamps are strings in form "2022-07-11T09:32:06,540". `Long.parseLong` will fail every time.

We could do something like that:
```
try {
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-ddThh:mm:ss,SSS");
    Date parsedDate = dateFormat.parse(instance.timestamp);
    return new LogEntry(entryProperties, new Timestamp(parsedDate));
} catch ...
```

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

**Comment by MMaiero** [comment]
Could you give it a try in this version? Thanks

**Review by marcellorinaldo — COMMENTED** [review]
Shall we set `DEFAULT_TIMESTAMP` to the current date? In my previous contribution I set it to `new Date(0)`

**Review by marcellorinaldo — APPROVED** [review]
Verified as working on Raspberry Pi 4

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**